
# HG changeset patch
# User Benjamin Peterson <benjamin@python.org>
# Date 1481183668 28800
# Node ID 4745d801cae2d57e3432313acd0b76b8b4cc9c75
# Parent  148c46d180b2a1510097a1eae79845b61190c879
guard HAVE_LONG_LONG definition to prevent redefinition (#28898)

diff --git a/Include/pyport.h b/Include/pyport.h
--- a/Include/pyport.h
+++ b/Include/pyport.h
@@ -37,9 +37,10 @@ Used in:  Py_SAFE_DOWNCAST
  * integral synonyms.  Only define the ones we actually need.
  */
 
-// long long is required now. Define HAVE_LONG_LONG unconditionally for
-// compatibility.
+// long long is required. Ensure HAVE_LONG_LONG is defined for compatibility.
+#ifndef HAVE_LONG_LONG
 #define HAVE_LONG_LONG
+#endif
 #ifndef PY_LONG_LONG
 #define PY_LONG_LONG long long
 /* If LLONG_MAX is defined in limits.h, use that. */

